docs/adr/009_service_account_users.md

```diff
@@ -0,0 +1,72 @@
+## Context and problem statement
+
+Cloud.gov provides a broker that allows users to create service accounts which can be used to manage or monitor applications programmatically. This service broker, the [UAA Credentials Broker](https://github.com/cloud-gov/uaa-credentials-broker), creates both a UAA and associated CAPI user for each service account key using a GUID for a username.
+
+These service account users are interspersed with human users in query results which can be confusing for humans tasked with managing or auditing users. We want to be able to mark service account users as being non-humans and help users understand an individual service account user's purpose and usage.
+
+## Decision drivers / forces
+
+User research into an early prototype of the user management page uncovered that Cloud.gov users would like to quickly discern which users are humans versus which are service accounts and how they are being used.
+
+Although there are potentially other non-human accounts in Cloud.gov which were created outside of the service broker, these are likely edge cases and are not the focus of our work.
+
+## Considered options
+1. Mark accounts in CAPI on creation
+1. Pull additional details from UAA about users via uaa-bot
+1. Make assumptions about users based on CAPI features
+1. Allow users to mark and rename users
+
+## Decision outcome
+
+Option 3: make assumptions about users based on CAPI features
+
+In the future as our application nears production, we should reinvestigate option 1 (marking accounts in CAPI) for a more robust solution.
+
+__Justification__ : determining if a username matches against a service credential binding is an additional API call, but this is the lightest lift that accomplishes our goal at this moment
+
+__Consequences__ : any interface that displays a username will now need to check if they are likely to be a non-human and if so, send another API call. Additionally there may be complications displaying a service key name in place of a username, such as to client side sorting / filtering / searching.
+
+### Success criteria
+
+- We can reliably mark users as human or service account related
+- We can display the service account key name to provide context about the purpose of the non-human user
+
+## Pros and cons of the options
+
+### Option 1: Mark accounts in CAPI on creation
+
+Alter the service broker code to add an annotation or label on a user marking them as service account related. Change either the username or presentation name from a GUID to the service account key name. This could be only for accounts moving forward or could involve bulk updating existing service account users.
+
+* `+` Simple change in service broker if only accommodating future accounts
+* `+` Reduces any ambiguity about source, no additional API calls required
+* `-` Requires Cloud.gov to update a service broker with few recent changes
+* `-` Bulk updates to previously created service brokers could have unintended consequences
```

Comment:
FWIW - I checked logs for the last week, and nobody has made a request to the CF API with both `label(s)` or `annotation(s)` and `user` in the request, so I *think* annotations/labels would be relatively safe to back-port.
Maybe a bigger pain point would be showing these users as recently updated, if anyone is monitoring changes on their users for security/compliance reasons.

Reply:
Yeah, I figure if we used a precisely named thing like `cg-user-svc-acct: true` we probably wouldn't run the risk of overwriting something, especially since only admins can update user objects.  I hadn't considered the recently updated aspect and what that might mean for users.  Want me to add that to the ADR?